lib/mpl_toolkits/mplot3d/proj3d.py

```diff
@@ -130,20 +130,19 @@ def ortho_transformation(zfront, zback):
 
 def proj_transform_vec(vec, M):
     vecw = np.dot(M, vec)
-    w = vecw[3]
-    # clip here..
-    txs, tys, tzs = vecw[0]/w, vecw[1]/w, vecw[2]/w
-    return txs, tys, tzs
+    vecw /= vecw[3]
+    return vecw[0:3]
 
 def proj_transform_vec_clip(vec, M):
     vecw = np.dot(M, vec)
-    w = vecw[3]
-    # clip here.
-    txs, tys, tzs = vecw[0] / w, vecw[1] / w, vecw[2] / w
+    # Determine clipping before rescaling
     tis = (0 <= vecw[0]) & (vecw[0] <= 1) & (0 <= vecw[1]) & (vecw[1] <= 1)
     if np.any(tis):
         tis = vecw[1] < 1
-    return txs, tys, tzs, tis
+    vecw /= vecw[3]
```

Comment:
This is not equivalent. The normalization is applied to the first three rows. Here, you do it to all four rows. This problem is only papered over by the subsequent assignment. You can just do `vecw[:3] /= vecw[3]` instead for 25% fewer division operations.